src/properties.ts

```diff
@@ -0,0 +1,317 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import fs from 'fs';
+import path from 'path';
+import slugify from 'slugify';
+import { version } from '../package.json';
+import {
+  DEFAULT_SONAR_EXCLUSIONS,
+  ENV_TO_PROPERTY_NAME,
+  ENV_VAR_PREFIX,
+  SCANNER_BOOTSTRAPPER_NAME,
+  SONAR_PROJECT_FILENAME,
+} from './constants';
+import { LogLevel, log } from './logging';
+import { ScanOptions } from './scan';
+import { ScannerProperties, ScannerProperty } from './types';
+
+/**
+ * Convert the name of a sonar property from its environment variable form
+ * (eg SONAR_SCANNER_FOO_BAR) to its sonar form (eg sonar.scanner.fooBar).
+ */
+function envNameToSonarPropertyName(envName: string) {
```

Comment:
```suggestion
function envNameToSonarPropertyNameMapper(envName: string) {
```
Include "mapper" to clearly indicate what we are doing with the values